### PR TITLE
added Rect type

### PR DESCRIPTION
**Pull request by aportner**

### CODE REVIEW

File: lib/types/Rect.lua [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+function metatable:__eq(other)
+	return self.Min == other.Min and self.Max == other.Max
+end
+
+function Rect.new(...)
+	if select("#", ...) == 4 then
+		local minX, minY, maxX, maxY = ...
+		if type(minX) ~= "number" or type(minY) ~= "number" or
+			type(maxX) ~= "number" or type(maxY) ~= "number" then
+			error("Vector2.new(minX, minY, maxX, maxY) takes in 4 numbers", 2)
```
Comment (LPGhatguy) [inline lib/types/Rect.lua]: Looks like you forgot to update this error message!

File: lib/types/Rect_spec.lua [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	it("should equal another Rect with the same values", function()
+		local r1 = Rect.new(10, 30, 50, 200)
+		local r2 = Rect.new(Vector2.new(10, 30), Vector2.new(50, 200))
+
+		assert.equals(r1, r1)
+		assert.equals(r1, r2)
+		assert.equals(r2, r2)
+	end)
+
+	it("should not equal another UDim with different scale and offset", function()
```
Comment (LPGhatguy) [inline lib/types/Rect_spec.lua]: `s/UDim/Rect/`

File: lib/types/Rect.lua [hunk 3]
```diff
@@ -0,0 +1,70 @@
+local assign = import("../assign")
+local typeof = import("../functions/typeof")
+local Vector2 = import("./Vector2")
+
+local Rect = {}
+
+setmetatable(Rect, {
+	__tostring = function()
+		return "Rect"
```
Comment (LPGhatguy) [inline lib/types/Rect.lua]: Looks like coverage is missing this line. We should make sure that `typeof` correctly picks up this type which should fix that.

**Comment by aportner** [comment]
fixed, added typeof coverage

**Pull request merged**